src/vsDebugServer.ts

```diff
@@ -0,0 +1,137 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+require('source-map-support').install(); // Enable TypeScript stack traces translation
+import 'reflect-metadata';
+
+/**
+ * This script launches vscode-js-debug in server mode for Visual Studio
+ */
+import * as net from 'net';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import { createGlobalContainer } from './ioc';
+import { IDebugSessionLike, ISessionLauncher, Session } from './sessionManager';
+import { getDeferred, IDeferred } from './common/promiseUtil';
+import DapConnection from './dap/connection';
+import { IPseudoAttachConfiguration } from './configuration';
+import { DebugConfiguration } from 'vscode';
+import { ServerSessionManager } from './serverSessionManager';
+import { ITarget } from './targets/targets';
+import { Readable, Writable } from 'stream';
+
+const storagePath = fs.mkdtempSync(path.join(os.tmpdir(), 'vscode-js-debug-'));
+
+class VSDebugSession implements IDebugSessionLike {
+  constructor(
+    public id: string,
+    name: string,
+    private readonly childConnection: Promise<DapConnection>,
+    public readonly configuration: DebugConfiguration,
+  ) {
+    this._name = name;
+  }
+
+  private _name: string;
+  set name(newName: string) {
+    this._name = newName;
+    this.childConnection
+      .then(conn => conn.initializedBlocker)
+      .then(conn => conn.dap())
+      .then(dap => {
+        dap.process({ name: newName });
+      });
+  }
+  get name() {
+    return this._name;
+  }
+}
+
+class VsDebugServer implements ISessionLauncher<VSDebugSession> {
+  private readonly sessionServer: ServerSessionManager<VSDebugSession>;
+
+  constructor(inputStream?: Readable, outputStream?: Writable) {
+    const services = createGlobalContainer({ storagePath, isVsCode: false });
+    this.sessionServer = new ServerSessionManager(services, this);
+
+    if (inputStream && outputStream) {
+      this.launchRootFromExisting(inputStream, outputStream);
+    } else {
+      this.launchRoot();
+    }
+  }
+
+  launchRootFromExisting(inputStream: Readable, outputStream: Writable) {
+    const deferredConnection: IDeferred<DapConnection> = getDeferred();
+    const session = new VSDebugSession(
+      'root',
+      'JavaScript debugger root session',
```

Comment:
Is this shown anywhere? Should we localize it?

Reply:
It's potentially shown in the process window and process drop-down in VS (though it isn't being shown right now). I'm actually inclined to just pass an empty string rather than trying to localize this, as it's more an implementation detail than something users care about.